postgres/matviews/sms_error_and_users_docs_replication.sql

```diff
@@ -0,0 +1,86 @@
+/* In this view we regroup the metrics of:
+  - sms_error_states
+  - users_replication_limit
+  - replication docs
+*/
+CREATE MATERIALIZED VIEW public.app_monitoring_sms_error_and_users_docs_replication
+TABLESPACE pg_default
+AS
+WITH sms_error_states AS (
+SELECT COALESCE(mu.partner_name, mu.url) AS partner_name,
+        date_trunc(
+            'day',
+            to_timestamp(
+                (
+                  (md.doc::jsonb#>>'{date,current}')::bigint / 1000
+                )::double precision
+            )
+        )::date AS reported,
+        COALESCE(
+            (md.doc::jsonb#>>'{messaging,outgoing,state,due}')::int,
+            0
+        ) as due,
+        COALESCE(
+            (
+                md.doc::jsonb#>>'{messaging,outgoing,state,muted}'
+            )::int,
+            0
+        ) as muted,
+        COALESCE(
+            (
+                md.doc::jsonb#>>'{messaging,outgoing,state,failed}'
+            )::int,
+            0
+        ) as failed,
+        COALESCE(
+            (
+                md.doc::jsonb#>>'{messaging,outgoing,state,delivered}'
+            )::int,
+            0
+        ) as delivered,
+        COALESCE(
+            (
+                md.doc::jsonb#>>'{messaging,outgoing,state,scheduled}'
+            )::int,
+            0
+        ) as scheduled
+    FROM (
+            SELECT url_id,
+                doc,
+                ROW_NUMBER() OVER (
+                    PARTITION BY url_id,
+                    date_trunc(
+                        'day',
+                        to_timestamp(
+                            (
+                                (doc::jsonb#>>'{date,current}')::bigint / 1000
+                            )::double precision
```

Comment:
if there are two rows with the `same day` and with `doctype = 'monitoring'` in monitoring_docs I believe doing a join like [here](https://github.com/medic/cht-app-monitoring-data-ingest/pull/48/files#diff-25ccf03359502816d961ad7b0dc2467113f78864a88208e0172c5aab7d39b6a1R79) will result in duplicate results in the final output

i believe this ROW_NUMBER() and `rank = 1` code is intended to prevent that. 
instead of using the `WITH` statement is it possible to use what is here and just add to the select

Reply:
alternatively you could decide it isn't needed because superset forces you to specify how aggregation is done when there are duplicate values...